README.md

```diff
@@ -8,9 +8,110 @@ Ubuntu Proxy Manager is a D-Bus mediated service that allows for managing system
 [![Go Report Card](https://goreportcard.com/badge/ubuntu/ubuntu-proxy-manager)](https://goreportcard.com/report/ubuntu/ubuntu-proxy-manager)
 [![License](https://img.shields.io/badge/License-GPL3.0-blue.svg)](https://github.com/ubuntu/ubuntu-proxy-manager/blob/main/LICENSE)
 
+## Installation
+
+Use APT to install the package on Ubuntu, which will set up all the required configuration files:
+
+```sh
+sudo apt install ubuntu-proxy-manager
+```
+
+## Usage
+
+The service currently exposes a single D-Bus method, `com.ubuntu.ProxyManager.Apply`, taking 6 string arguments:
+- `http` - HTTP proxy
+- `https` - HTTPS proxy
+- `ftp` - FTP proxy
+- `socks` - SOCKS proxy
+- `no_proxy` - hosts excluded from proxy
+- `auto` - proxy autoconfiguration URL
+
+When calling the function, all 6 arguments must be passsed. Arguments can be skipped by replacing them with empty strings. Keep in mind that this function is not additive and it replaces previously set proxy settings on each call.
+
+``` sh
+# Only apply HTTP proxy
+gdbus call --system --dest com.ubuntu.ProxyManager \
+                    --object-path /com/ubuntu/ProxyManager \
+                    --method com.ubuntu.ProxyManager.Apply \
+                    "http://example.com:8080" "" "" "" "" ""
+
+# Only set proxy autoconfiguration URL, HTTP proxy is removed
+gdbus call --system --dest com.ubuntu.ProxyManager \
+                    --object-path /com/ubuntu/ProxyManager \
+                    --method com.ubuntu.ProxyManager.Apply \
+                    "" "" "" "" "" "http://example.com:8080/proxy.pac"
+
+# Remove all previously set proxy settings
+gdbus call --system --dest com.ubuntu.ProxyManager \
+                    --object-path /com/ubuntu/ProxyManager \
+                    --method com.ubuntu.ProxyManager.Apply \
+                    "" "" "" "" "" ""
+```
+
+Due to the privileged nature of the service, polkit authorization is set in place to only allow admins to execute the `Apply` method.
+
+Some backends do not support all configuration options. These are described below and will be silently skipped on proxy application.
+
+### Proxy URL format
+
+For the individual proxy protocols, the URL must be in the form of:
+
+```
+protocol://username:password@host:port
+```
+
+It is not mandatory to escape special characters in the username or password. The service will escape any unescaped special character before applying the proxy settings, and will take care not to double-escape already escaped characters.
+
+### `no_proxy` format
+
+The host exclusion setting must be in the form of:
+
+```
+localhost,127.0.0.1,::1
+```
+
+Hosts can be individually wrapped in single (`'`) or double quotes (`"`), or separated by spaces.
+
+## Support
```

Comment:
Having this section named as `Support` is a bit weird as it can indicate tons of different stuff, maybe we should go for something like `Supported Configurations`?